src/Compilers/VisualBasic/Portable/Parser/ParseConditional.vb

```diff
@@ -248,7 +251,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
         Private Function ParseRegionDirective(hashToken As PunctuationSyntax) As RegionDirectiveTriviaSyntax
             Debug.Assert(CurrentToken.Kind = SyntaxKind.IdentifierToken AndAlso DirectCast(CurrentToken, IdentifierTokenSyntax).PossibleKeywordKind = SyntaxKind.RegionKeyword,
-                         "ParseRegionDirective called with wrong token.")
+                         $"{NameOf(ParseRegionDirective)} called with wrong token.")
```

Comment:
Do you feel that string concatenation would be clearer?  Otherwise, I'm not sure the exact form of the assert message is important.